pkg/sql/parser/sql.y

```diff
@@ -5046,14 +5065,204 @@ show_stmt:
 | show_full_scans_stmt
 | show_default_privileges_stmt // EXTEND WITH HELP: SHOW DEFAULT PRIVILEGES
 
-// Cursors are not yet supported by CockroachDB. CLOSE ALL is safe to no-op
-// since there will be no open cursors.
+// %Help: CLOSE - close SQL cursor
+// %Category: Misc
+// %Text: CLOSE [ ALL | <name> ]
+// %SeeAlso: DECLARE, FETCH
 close_cursor_stmt:
-	CLOSE ALL { }
-| CLOSE cursor_name { return unimplementedWithIssue(sqllex, 41412) }
+	CLOSE ALL
+	{
+	  $$.val = &tree.CloseCursor{
+	    All: true,
+	  }
+	}
+| CLOSE cursor_name
+  {
+	  $$.val = &tree.CloseCursor{
+	    Name: tree.Name($2),
+	  }
+	}
+| CLOSE error // SHOW HELP: CLOSE
 
+// %Help: DECLARE - declare SQL cursor
+// %Category: Misc
+// %Text: DECLARE <name> [ options ] CURSOR p [ WITH | WITHOUT HOLD ] FOR <query>
+// %SeeAlso: CLOSE, FETCH
 declare_cursor_stmt:
-	DECLARE { return unimplementedWithIssue(sqllex, 41412) }
+	DECLARE cursor_name opt_binary opt_sensitivity opt_scroll CURSOR opt_hold FOR select_stmt
```

Comment:
i think these cursor_options can be in any order in PG, can we match?

https://github.com/postgres/postgres/blob/8a2e323f208557acbfdc911516e84ec017f5a6ca/src/backend/parser/gram.y#L11437

Reply:
I'm not sure the best way to do this.

Reply:
can we do something like

```
cursor_option:
    BINARY { BinaryCursorOption } 
|   ...

opt_cursor_options:
   cursor_options { return $1 }
| /* blank */ { return []CursorOption{} }

cursor_options:
    cursor_option { []CursorOption{...} }
    cursor_options ',' cursor_option { return append($1, $3) }
```

And have `CursorOption` implement an interface `ApplyCursorOption` which applies some sort of method onto the `Cursor` AST node?

you can leave it as a TODO, not too fussed

Reply:
I left a TODO 😅